### PR TITLE
Modify the CI to execute on 1.64.0

### PR DESCRIPTION
**Pull request by YangKeao**
```
error: package `addr2line v0.21.0` cannot be built because it requires rustc 1.65 or newer, while the currently active rustc version is 1.64.0
```

After patching the minor version of `pprof-rs`, it can actually run on 1.64.0

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]

**Review by Xuanwo — APPROVED** [review]

**Comment by sticnarf** [comment]
As I replied in #221, addr2line 0.21 does not force us to bump MSRV to 1.65 because it's possible to keep the patch version of `backtrace` low enough.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -35,6 +35,13 @@ jobs:
       - name: Remove pre-generated prost files to force regeneration
         run: rm proto/*.rs
 
+      - name: Set precise version to allow building on 1.64.0
```
Comment (Xuanwo) [inline .github/workflows/rust.yml:38]: How about commit `Cargo.lock` into the repo so that developers don't need to figure all of those?

**Review by Xuanwo — APPROVED** [review]

File: .github/workflows/rust.yml (hunk 1 above)
Comment (YangKeao) [reply .github/workflows/rust.yml]: Nice suggestion. I've read https://blog.rust-lang.org/2023/08/29/committing-lockfiles.html and https://github.com/rust-lang/cargo/commit/3bb02f23a23f90261b44a107e1b8821a504bac08. It seems good to just commit the `.lock` file 🍻.

**Comment by YangKeao** [comment]
> As I replied in #221, addr2line 0.21 does not force us to bump MSRV to 1.65 because it's possible to keep the patch version of `backtrace` low enough.

Thanks for the information. Then I'll not bump the MSRV in this PR, but just commit the `Cargo.lock` and make sure the CI passes. PTAL @sticnarf

**Review by sticnarf — APPROVED** [review]
LGTM

**Pull request merged**